vm/src/machines.rs

```diff
@@ -6,35 +6,56 @@
 use super::{memory::Memory, rv32::SOP};
 use crate::{NexusVM, Regs};
 
+type TestMachine<'a> = (&'a str, fn() -> Vec<u32>, fn() -> Regs, fn() -> Vec<u8>);
+
 /// An array of test machines, useful for debugging and developemnt.
 #[allow(clippy::type_complexity)]
-pub const MACHINES: &[(&str, fn() -> Vec<u32>, fn() -> Regs)] = &[
-    ("nop10", || nop_code(10), || nop_result(10)),
-    ("loop10", || loop_code(10), || loop_result(10)),
-    ("fib31", fib31_code, fib31_result),
-    ("bitop", bitop_code, bitop_result),
-    ("branch", branch_code, branch_result),
-    ("jump", jump_code, jump_result),
-    ("ldst", ldst_code, ldst_result),
-    ("shift", shift_code, shift_result),
-    ("sub", sub_code, sub_result),
+pub const MACHINES: &[TestMachine] = &[
+    ("nop10", || nop_code(10), || nop_result(10), || [].to_vec()),
```

Comment:
Probably a bit cleaner to do these as `vec![]`.

Reply:
When I applied this change, clippy said `|| vec![]` is `Vec::new` so I followed that.